CFile.js

```diff
@@ -0,0 +1,229 @@
+/*
+ * CFile.js - plugin to extract resources from a C source code file
+ *
+ * Copyright © 2019, JEDLSoft
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+var fs = require("fs");
+var path = require("path");
+var log4js = require("log4js");
+
+var logger = log4js.getLogger("loctool.plugin.CFile");
+
+/**
+ * Create a new java file with the given path name and within
+ * the given project.
+ *
+ * @param {Project} project the project object
+ * @param {String} pathName path to the file relative to the root
+ * of the project file
+ * @param {FileType} type the file type of this instance
+ */
+var CFile = function(props) {
+    this.project = props.project;
+    this.pathName = props.pathName;
+    this.type = props.type;
+    this.API = props.project.getAPI();
+
+    this.set = this.API.newTranslationSet(this.project ? this.project.sourceLocale : "zxx-XX");
+};
+
+/**
+ * Unescape the string to make the same string that would be
+ * in memory in the target programming language.
+ *
+ * @static
+ * @param {String} string the string to unescape
+ * @returns {String} the unescaped string
+ */
+CFile.unescapeString = function(string) {
+    var unescaped = string;
+
+    unescaped = unescaped.
+        replace(/\\\\n/g, "").                // line continuation
```

Comment:
I don't think C support line continuation, does it?

Reply:
I removed it. When making a key, We don't manipulate string at all. Regarding that, I added more test cases. String with a different number of spaces will be handled as a different key each. https://github.com/iLib-js/ilib-loctool-webos-c/blob/initialSetup/test/testCFile.js#L83